File: hash-tables-cherelemma-main/challenge.js
function hashTable(arr) {
  // write your code here
  const hashTable = Array.from({length: 11}, () => []);
    for (let num of arr) {
      let hash = Math.abs(num) % 11;
      hashTable[hash].push(num);
  }
  return hashTable.flat();
}

console.log(hashTable([12, 24, 125, 5, 91, 1106, 2, 1021, 29, 3536, 10]))
// => [12, 24, 2, 91, 125, 5, 3536, 1106, 29, 1021, 10]

console.log(hashTable([2, 341, 73, 8265, 234004, 602, 7400000, 200000000]))
// => [341, 234004, 2, 200000000, 7400000, 8265, 73, 602]

module.exports = hashTable

File: Challenges with Solutions/insertion-sort-intro-cherelemma-main/challenge.rb
def insertion_sort(array)
  # write your code here
  index = array.length - 1
  valueToInsert = array[index]
  while (index >= 0)
    if index != 0 && valueToInsert < array[index - 1]
      array[index] = array[index - 1]
      puts array.join(' ')
      index -= 1
    elsif
      array[index] = valueToInsert
      puts array.join(' ')
      break
    end
  end
end

insertion_sort([1, 3, 2])
# => 1 3 3
#    1 2 3

puts '-' * 20

insertion_sort([1, 4, 6, 9, 3])
# => 1 4 6 9 9
#    1 4 6 6 9
#    1 4 4 6 9
#    1 3 4 6 9
